src/nvim/getchar.c

```diff
@@ -1847,7 +1849,7 @@ static mapblock_T *find_typed_map(const bool timedout, const int local_State,
     return NULL;
   }
 
-  int keylen = *keylenp;
+  int keylen = 0;
   mapblock_T *mp = NULL;
   mapblock_T *mp2 = NULL;
   int nolmaplen;
```

Comment:
Can't comment below, but at the end of the function I see:

```
  // If no partly match found, use the longest full
  // match.
  if (keylen != KEYLEN_PART_MAP) {
    mp = mp_match;
    keylen = mp_match_len;
  }
  *keylenp = keylen;
  return mp;
```

```
// If no partly match found, use the longest full match.
if (keylen != KEYLEN_PART_MAP) {
  return mp_match;
}
*keylenp = keylen;
return mp;
```

While rewriting this I realized that `keylen = mp_match_len` is a completely unused statement (don't we have warnings for this?).

The fact that this exists tells me fishy stuff may be going on.